diffrax/_root_finder/_verychord.py

```diff
@@ -94,11 +94,12 @@ def init(
                 y_dtype = lxi.default_floating_dtype()  # pyright: ignore
             else:
                 y_dtype = jnp.result_type(*y_leaves)
+            diff_dtype = jnp.finfo(y_dtype).dtype
```

Comment:
I think if this trick is going to be appearing in a few places then it should probably be pulled out into a `complex_to_real_dtype` function in `_misc.py`. This will improve readability by giving a name to this operation.